addons/hr_holidays/models/hr_leave.py

```diff
@@ -1104,11 +1105,13 @@ def action_approve(self):
 
         # Post a second message, more verbose than the tracking message
         for holiday in self.filtered(lambda holiday: holiday.employee_id.user_id):
+            user_tz = timezone(self.env.user.tz if self.env.user.tz else 'UTC')
```

Comment:
You can use the timezone from the leave directly :-)

```suggestion
            user_tz = timezone(holiday.tz)
```

Reply:
> You can use the timezone from the leave directly :-)

tkanks !